packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -556,22 +556,24 @@ function recursivelyDisconnectChildren(vnodes: VNodes) {
     }
 }
 
-// This is a super optimized mechanism to remove the content of the shadowRoot without having to go
+// This is a super optimized mechanism to remove the content of the root node (shadow root
+// for shadow DOM components and the root element itself for light DOM) without having to go
 // into snabbdom. Especially useful when the reset is a consequence of an error, in which case the
 // children VNodes might not be representing the current state of the DOM.
-export function resetShadowRoot(vm: VM) {
-    const { children, cmpRoot, renderer } = vm;
+export function resetComponentRoot(vm: VM) {
+    const { children, cmpRoot, renderer, elm } = vm;
+    const rootNode = isNull(cmpRoot) ? elm : cmpRoot;
```

Comment:
Are we not using the `hasShadow` helper to avoid another property lookup?

Reply:
Can we add a getter on the VM that gives us the rootNode? We are getting the rootNode this way in a couple of places.

Reply:
Yes adding a `getRenderRoot` helper to `VM.ts` would make sense.

Reply:
Sure, I will add it.

Reply:
Done.